### PR TITLE
Missile(defense) improvements...

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
...And some minor tank round stuff
-Tweak Abrams Turret accuracy
-GM 'no Weapon' kill now looks for no weapons with ammo, instead of no weapons
-Make Weave Strength editable for VTOL AI
-Add GPS jamming functionality for jamming continuously-updating GPS munitions
-AI will toggle/disable jammers if incoming GPS/Antirad ordinance
-Add Inertial Guidance mode for missiles
-Tweak AI launch behavior when launching radar missiles without a radar(lock) to wait until within missile's self lockon range
-Rework missile interception, missiles set to engageMissiles = true will now autolaunch against incoming missiles (launch geometry permitting)
-APS now accounts for if individual turrets are reloading/overheated when distributing targets.
-APS will only assign single interceptor turret to incoming shells; rapid fire APS relying on weight of fire instead of precision will still assign multiple turrets if there are more turrets than targets
-Fix Antirads set to home in on ECM jammers not homing.
-Single-stage Modular missiles now auto-ignite motor(s) on launch if no launch AG setup.

### CODE REVIEW

File: BDArmory/Weapons/Missiles/MissileBase.cs [hunk 1]
```diff
@@ -595,25 +599,34 @@ public Vector3d UpdateGPSTarget()
                 gpsTargetCoords_ = targetGPSCoords;
                 if (targetVessel && HasFired && (gpsUpdates >= 0f))
                 {
-                    var weaponManager = VesselModuleRegistry.GetMissileFire(SourceVessel);
-                    if (weaponManager != null && weaponManager.CanSeeTarget(targetVessel, false))
+                    float distanceToTarget = (vessel.transform.position - gpsTargetCoords_).sqrMagnitude;
```
Comment (BrettRyland) [inline BDArmory/Weapons/Missiles/MissileBase.cs:602]: Maybe rename `distanceToTarget` to `distanceToTargetSqr` for readability?

File: BDArmory/Weapons/Missiles/MissileBase.cs [hunk 2]
```diff
@@ -1156,6 +1169,33 @@ protected void UpdateAntiRadiationTarget()
...
                 TargetPosition = VectorUtils.GetWorldSurfacePostion(targetGPSCoords, vessel.mainBody);
         }
+        private bool setInertialTarget = false;
+
+        public Vector3d UpdateInertialTarget()
+        {
+            Vector3 TargetCoords_;
+            if (!setInertialTarget)
+            {
+                driftSeed = new Vector3(UnityEngine.Random.Range(-1, 1) * inertialDrift, UnityEngine.Random.Range(-1, 1) * inertialDrift, UnityEngine.Random.Range(-1, 1) * inertialDrift);
```
Comment (BrettRyland) [inline BDArmory/Weapons/Missiles/MissileBase.cs:1179]: `inertialDrift * Random.insideUnitSphere` (https://docs.unity3d.com/ScriptReference/Random-insideUnitSphere.html) might be more appropriate for this.

File: BDArmory/Weapons/Missiles/BDModularGuidance.cs [hunk 3]
```diff
@@ -1024,7 +1024,7 @@ public static PartModule FindFirstDecoupler(Part parent, PartModule last)
 
         /// <summary>
         ///     This method will execute the next ActionGroup. Due to StageManager is designed to work with an active vessel
-        ///     And a missile is not an active vessel. I had to use a different way handle stages. And action groups works perfect!
+        ///     And a missile is not an active vessel. I had to use a different way to handle stages. And action groups works perfect!
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/Weapons/Missiles/BDModularGuidance.cs]: It is a usage that is a tad archaic, yes.
And as a result, fallen out of vogue.

File: BDArmory/Control/MissileFire.cs [hunk 4]
```diff
@@ -2995,7 +2995,7 @@ IEnumerator AllCMRoutine(int count)
             for (int i = 0; i < count; ++i)
             {
                 DropCMs((int)(CMDropper.CountermeasureTypes.Flare | CMDropper.CountermeasureTypes.Chaff | CMDropper.CountermeasureTypes.Smoke));
-                if (i < count - 1) // Don't wait on the last one.
+                if (i <= count) // Don't wait on the last one.
```
Comment (BrettRyland) [inline BDArmory/Control/MissileFire.cs:2998]: This one shouldn't be changed. Its purpose was to not wait after dropping the last CM.

Comment (SuicidalInsanity) [reply BDArmory/Control/MissileFire.cs]: fixed

**Pull request merged**